packages/cli/generators/model/index.js

```diff
@@ -154,20 +154,43 @@ module.exports = class ModelGenerator extends ArtifactGenerator {
       },
     ])
       .then(props => {
-        if (typeof props.modelBaseClass === 'object')
+        if (typeof props.modelBaseClass === 'object') {
           props.modelBaseClass = props.modelBaseClass.value;
+        }
 
         Object.assign(this.artifactInfo, props);
         debug(`props after model base class prompt: ${inspect(props)}`);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during model base class prompt: ${err}`);
+        return this.exit(err);
+      });
+  }
+
+  async promptStrictMode() {
+    if (this.shouldExit()) return false;
+    return this.prompt([
+      {
+        name: 'disabledStrictMode',
+        message: 'Disable strict mode for this model?',
+        type: 'confirm',
+        default: false,
+        when: !this.artifactInfo.disabledStrictMode,
+      },
+    ])
+      .then(setting => {
```

Comment:
Correct me if wrong:
I assume the setting should be an object with the `disabledStrictMode` property like:
`{disabledStrictMode: true}`

I prefer to add a property called `modelSettings` in `this.artifactInfo` in case we support more configurable setting attributes in the future, like:

```ts
artifactInfo: {
  modelSettings: {
    disabledStrictMode: true
  }
}
```

And btw, why do we return the `setting` at the end of the prompt 🤔

Reply:
Thanks, I'll update it accordingly and remove the return!